app/main-chain-hash.py

```diff
@@ -0,0 +1,150 @@
+from typing import Tuple, Any, Iterable, Union
+
+
+# Chain Hash
+class Dictionary:
+    def __init__(self,
+                 initial_capacity: int = 8,
+                 load_factor: float = 0.75) -> None:
+        self.table: list[list[Tuple[Any, Any]]] = \
+            [[] for _ in range(initial_capacity)]
+        self.size: int = 0
+        self.load_factor = load_factor
+        self.capacity = initial_capacity
+
+    def _hash(self, key: Any) -> int:
+        return hash(key) % self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        index = self._hash(key)
+        bucket = self.table[index]
+
+        for i, (k, v) in enumerate(bucket):
+            if k == key:
+                bucket[i] = (key, value)
+                return
+
+        bucket.append((key, value))
+        self.size += 1
+
+        if self.size / self.capacity > self.load_factor:
+            self._resize()
+
+    def __getitem__(self, key: Any) -> Any:
+        index = self._hash(key)
+        bucket = self.table[index]
+
+        for k, v in bucket:
+            if k == key:
+                return v
+        raise KeyError(f"Key '{key}' not found.")
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __delitem__(self, key: Any) -> None:
+        index = self._hash(key)
+        bucket = self.table[index]
+
+        for k, v in bucket:
+            if k == key:
+                bucket.remove((k, v))
+                self.size -= 1
+                return
+
+        raise KeyError(f"Key '{key}' not found.")
+
+    def __iter__(self) -> Iterable[Any]:
+        for bucket in self.table:
+            for key in bucket:
+                yield key
```

Comment:
The `__iter__` method currently yields keys instead of key-value pairs. If the intention is to iterate over keys, this is correct. However, if you want to iterate over key-value pairs, consider yielding `(key, value)` instead.